Sources/SwiftDocC/Checker/Checkers/SeeAlsoInTopicsHeadingChecker.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  SeeAlsoInTopicsHeadingChecker.swift
+//  SeeAlsoInTopicsHeadingChecker
+//
+//  Created by Kyle on 2021/10/16.
+//
+
+import Foundation
+import Markdown
+
+public struct SeeAlsoInTopicsHeadingChecker: Checker {
+    public var topicsHeading: Heading?
+    public var seeAlsoInTopicsHeadings: [Heading] = []
+    public var problems: [Problem] {
+        guard !seeAlsoInTopicsHeadings.isEmpty else {
+            return []
+        }
+        return seeAlsoInTopicsHeadings.compactMap { heading -> Problem? in
+            guard let headingRange = heading.range else { return nil }
+            let diagnostic = Diagnostic(
+                source: sourceFile,
+                severity: .warning,
+                range: headingRange,
+                identifier: "org.swift.docc.SeeAlsoInTopicsHeadings",
+                summary: #"It's easy to accidentally write a "See Also" section as part of a Topics section instead of as it's own second-level heading like DocC expects"#
+            )
+            let replacement = Replacement(range: headingRange, replacement: "## \(heading.plainText)")
+            let solution = Solution(
+                summary: "Change the heading to a level-2 heading",
+                replacements: [replacement]
+            )
+            return Problem(diagnostic: diagnostic, possibleSolutions: [solution])
+        }
+    }
+
+    private var sourceFile: URL?
+
+    public init(sourceFile: URL?) {
+        self.sourceFile = sourceFile
+    }
+
+    public mutating func visitHeading(_ heading: Heading) {
+        if heading.level == 2, heading.plainText == "Topics" {
+            topicsHeading = heading
+            return
+        } else if topicsHeading != nil, heading.level > 2, heading.plainText == "See Also"{
+            seeAlsoInTopicsHeadings.append(heading)
+            return
+        }
```

Comment:
I expect that most developers who customize the See Also section would already have a Topics section on that page, so I think that warning on a “See Also” subheading under a Topics section would be the most common case.

That said, if the developer wrote a third-level section named “See Also” I’d think that they likely meant for that to form a See Also section.

I’m happy with this covering the common case (like this checker is already doing) but if you want to you can expand this check to warn about any third-level heading named “See Also”.

There are more places where it could make sense for Topics sections and See Also sections to be checked the same (for example warning about duplicate See Also sections or warning about third-level headings named “Topics”) but those can be future enhancements (I can report separate issues for those)